pkg/db/sqlite/sqlite.go

```diff
@@ -462,12 +462,12 @@ func (c *Client) queryHeaders(
 			return fmt.Errorf("could not prepare statement: %v", err)
 		}
 		defer reqHeadersStmt.Close()
-		for i := range reqLogs {
-			headers, err := findHeaders(ctx, reqHeadersStmt, reqLogs[i].ID)
+		for _, reqLog := range reqLogs {
+			headers, err := findHeaders(ctx, reqHeadersStmt, reqLog.ID)
 			if err != nil {
 				return fmt.Errorf("could not query request headers: %v", err)
 			}
-			reqLogs[i].Request.Header = headers
+			reqLog.Request.Header = headers
```

Comment:
Manually tested this, and the retrieved request headers aren't included in the API response. I believe it's because `reqLog.Request` is a value and not a pointer, so the original `reqLogs` that is returned upstream won't be mutated. But because `http.Header`'s underlying type is a `map` (which holds a reference), we could keep the `for _, reqLog` (I agree it's more idiomatic) and fill (copy) the found headers like so:   
```suggestion
			for key, value := range headers {
				reqLog.Request.Header[key] = value
			}
```
 Curious to hear your thoughts @holiman.

Reply:
Hm, maybe it would be better to change the method signature (and really, in general) to use pointers instead? So you'd use `reqLogs []*reqlog.Request`. I find that often makes it easier to reason about

Reply:
Your proposed variant uses the fact that the copied `Request` has copied the same pointer to a `map`, and thus you access the same underlying map. So mutating the original headers-map although the caller passed a copy. It's a bit of a hacky solution to the problem.

Reply:
I've changed the PR to now be only that change. However, since the egress-object is by-value (only the ingress `Response`  field is a reference), it means that none of the request headers become set. So this PR fixes the nil deref-panic, but doesn't fix the request headers.

Reply:
> However, since the egress-object is by-value (only the ingress `Response` field is a reference), it means that none of the request headers become set. So this PR fixes the nil deref-panic, but **doesn't fix the request headers**.

Are you sure about the request headers not being set? Just tested this manually, and the request headers are getting set. While indeed the `Request` field on the `reqlog.Request` struct is a value and not a pointer, because we're using the _index_ when ranging ([ref](https://github.com/holiman/hetty/blob/f975d1ce65565028e5e6d474e40813e0d7ba1dbd/pkg/db/sqlite/sqlite.go#L465)) and setting the headers via `reqLogs[i].Request.Header` ([ref](https://github.com/holiman/hetty/blob/f975d1ce65565028e5e6d474e40813e0d7ba1dbd/pkg/db/sqlite/sqlite.go#L470)), we're actually mutating the items referenced by the slice.

Reply:
> Are you sure about the request headers not being set? Just tested this manually,

You're right. I tested it too, but apparently I tested against the wrong binary.